### ISSUE TITLE
wc.Attribute.GetAll() method doesn't seem to accept parameters for pagination...

### ISSUE THREAD

**Issue by christechno25** [XiaoFaye/WooCommerce.NET#323]
Hello. I have this code...

` Dim dic As Dictionary(Of String, String) = New Dictionary(Of String, String)()`
        `dic.Add("per_page", "100")`
        `Dim pageNumber As Integer = 1`
        `dic.Add("page", pageNumber.ToString())`
        `Dim attribs As List(Of WooCommerceNET.WooCommerce.v2.ProductAttribute) = New List(Of WooCommerceNET.WooCommerce.v2.ProductAttribute)()`
        `Dim endWhile As Boolean = False`
       

`While Not endWhile`
            `Dim attribsTemp = Await wc.Attribute.GetAll(dic)`
            `MsgBox(attribsTemp.Count)`
            `If attribsTemp.Count > 0 Then`
                `attribs.AddRange(attribsTemp)`
                `pageNumber += 1`
               ` dic("page") = pageNumber.ToString()`
            `Else`
                `endWhile = True`
            `End If`
        `End While`

Basically it loops through pages until it reaches to a point where a page already contains 0 items, and at which point it has looped through all available items. This is a modified version of the retrieval of all products. It works but for some reason, it doesn't work when I retrieve all Attributes. I have put a message box to output the number it got and it always shows 20 regardless of the per_page and page parameters provided. Currently I only have 20 sample attributes but shouldn't it output 0 if I make it start retrieving at page 2 since I only have 20? Is this a normal behavior? Thank you very much.

**Comment by XiaoFaye** [comment]
I'm not sure if per_page and page will work for ProductAttribute, as it doesn't save in the database as a post.

**Comment by XiaoFaye** [comment]
Found it, it only accepts one parameter:

![image](https://user-images.githubusercontent.com/6334728/48165766-79c7f500-e34b-11e8-9dcc-e17e8ffdafee.png)

**Comment by XiaoFaye** [comment]
For now, Yes!

**Comment by JannikWorkAccount** [comment]
GetAll only retrieves the first 10 products. How can i get the next products?

**Comment by JannikWorkAccount** [comment]
This worked fine for me. 
`            Dictionary<string, string> dict = new Dictionary<string, string>();
            d.Add("per_page", "32");
            var products = await wc.Product.GetAll(d);`
